scene/gui/control.cpp

```diff
@@ -3049,24 +3051,17 @@ void Control::_notification(int p_notification) {
 			if (data.RI) {
 				get_viewport()->_gui_remove_root_control(data.RI);
 				data.RI = nullptr;
+				get_parent()->disconnect(SNAME("children_changed"), callable_mp(get_viewport(), &Viewport::gui_set_root_order_dirty));
 			}
 
 			data.parent_canvas_item = nullptr;
 			data.is_rtl_dirty = true;
 		} break;
 
-		case NOTIFICATION_MOVED_IN_PARENT: {
+		case NOTIFICATION_CHILDREN_CHANGED: {
 			// Some parents need to know the order of the children to draw (like TabContainer),
 			// so we update them just in case.
```

Comment:
This comment is no longer relevant.

Reply:
I kept it because that's what it is about, it is calling queue_redraw() for the very reason the comment describes.